src/main/java/bio/terra/janitor/service/cleanup/FlightScheduler.java

```diff
@@ -0,0 +1,90 @@
+package bio.terra.janitor.service.cleanup;
+
+import bio.terra.janitor.app.configuration.PrimaryConfiguration;
+import bio.terra.janitor.db.JanitorDao;
+import bio.terra.janitor.service.stairway.StairwayComponent;
+import com.google.common.base.Preconditions;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** The FlightScheduler runs the {@link FlightManager} periodically to clean resources. */
+// TODO add metrics.
+@Component
+public class FlightScheduler {
+  /** How often to query for flights to schedule. */
+  private static final Duration SCHEDULE_PERIOD = Duration.ofMinutes(1);
```

Comment:
Just curious how we get this number? Seems the schedule runs every one minute and do the `scheduleFlights`. And the `scheduleFlights` will keep do things until empty.  Will scheduleFlights usually takes longer than 1 minute to finish and we just want the schedule things last forever with no delay? Or we want to give it a bigger number than the normal one `scheduleFlights` run?

Reply:
I do not know what the right numbers are here, and they'll probably change with different workloads. No idea yet how long this will typically take. I don't think we want to look for new resources more than once a minute, so that seemed like an ok place to start.
If scheduleFlights does not complete in less than a minute, a new scheduleFlights will not be started by the ScheduledExecutorService until the previous call completes. So we don't have to worry about them running simultaneously. Adding a comment.